diff --git a/win32/Makefile.sub b/win32/Makefile.sub
index 326bd7fa7a..1d1bc0385a 100644
--- a/win32/Makefile.sub
+++ b/win32/Makefile.sub
@@ -262,7 +262,7 @@ CFLAGS = $(CFLAGS_NO_ARCH) $(ARCH_FLAG)
 CXXFLAGS = $(CFLAGS)
 !endif
 !if !defined(LDFLAGS)
-LDFLAGS = -incremental:no -debug -opt:ref -opt:icf
+LDFLAGS = -incremental:no -debug -opt:ref -opt:icf -ignore:4099
 !endif
 !if !defined(XLDFLAGS)
 XLDFLAGS = -stack:$(STACK)
@@ -1152,7 +1152,7 @@ $(LIBRUBY_SO):	$(LIBRUBY_A) $(DLDOBJS) $(RUBYDEF) $(RUBY_SO_NAME).res
 		@-$(PRE_LIBRUBY_UPDATE)
 !endif
 		$(ECHO) linking shared-library $(@:\=/)
-		$(Q) $(LDSHARED) $(DLDOBJS) $(LIBRUBY_A) \
+		$(Q) $(LDSHARED) $(MAINOBJ) $(DLDOBJS) $(LIBRUBY_A) \
 			$(RUBY_SO_NAME).res $(SOLIBS) $(EXTSOLIBS) $(LIBS) -Fe$@ -link $(LDFLAGS) \
 			$(LIBRUBY_DLDFLAGS)
 		@$(RM) dummy.lib dummy.exp
