diff --git a/ext/fiddle/extconf.rb b/ext/fiddle/extconf.rb
index f8a94d4..05340ca 100644
--- a/ext/fiddle/extconf.rb
+++ b/ext/fiddle/extconf.rb
@@ -10,6 +10,12 @@ if ! bundle
   pkg_config("libffi") and
     ver = pkg_config("libffi", "modversion")
 
+  if $mswin
+    $defs << "-DFFI_BUILDING"
+    CONFIG['CFLAGS'] << " -DFFI_BUILDING"
+    $CFLAGS << " -DFFI_BUILDING"
+  end
+
   if have_header(ffi_header = 'ffi.h')
     true
   elsif have_header(ffi_header = 'ffi/ffi.h')
diff --git a/win32/Makefile.sub b/win32/Makefile.sub
index 326bd7f..c080e53 100644
--- a/win32/Makefile.sub
+++ b/win32/Makefile.sub
@@ -380,7 +380,7 @@ MINIRUBY = $(RUBY) -I$(MAKEDIR) -r$(arch)-fake
 RUNRUBY = $(MINIRUBY)
 !else
 MINIRUBY = .\miniruby$(EXEEXT) -I$(srcdir)/lib -I.
-RUNRUBY = .\$(PROGRAM) -I$(srcdir)/lib -I"$(EXTOUT)/$(arch)" -I.
+RUNRUBY = .\$(PROGRAM) --disable-gems -I$(srcdir)/lib -I"$(EXTOUT)/$(arch)" -I.
 !endif
 MINIRUBY = $(MINIRUBY) $(MINIRUBYOPT)
 RUNRUBY = $(RUNRUBY) "$(srcdir)/tool/runruby.rb" --extout="$(EXTOUT)" $(RUNRUBYOPT) -- $(RUN_OPTS)
